src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -140,50 +125,54 @@ public bool TryCreateSolutionContext(out IVsPathContext outputPathContext)
         }
 
 
-        private static async Task<Dictionary<string, EnvDTE.Project>> GetPathToDTEProjectLookupAsync(EnvDTE.DTE dte)
+        private async Task<Tuple<IVsProjectAdapter, string>> CreateProjectAdapterAsync(string projectUniqueName)
         {
             await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            var pathToProject = new Dictionary<string, EnvDTE.Project>(StringComparer.OrdinalIgnoreCase);
-
+            var dte = _serviceprovider.GetDTE();
             var supportedProjects = dte.Solution.Projects.Cast<EnvDTE.Project>();
 
             foreach (var solutionProject in supportedProjects)
             {
                 var solutionProjectPath = EnvDTEProjectInfoUtility.GetFullProjectPath(solutionProject);
 
                 if (!string.IsNullOrEmpty(solutionProjectPath) &&
-                    !pathToProject.ContainsKey(solutionProjectPath))
+                    solutionProjectPath.Equals(projectUniqueName))
```

Comment:
use ordinalignorecase string comparer

Reply:
it was intentional to use ordinal, but I guess I can use `OSBasedStringComparer` instead

Reply:
while using OSBasedStringComparer is fine, i am wondering in which case would this ever need a case sensitive comparison, given that this API is for VS only, and VS only runs on windows.